### PR TITLE
Add operator-framework-catalogd

### PR DESCRIPTION
**Pull request by ncdc**

### CODE REVIEW

**Comment by ncdc** [comment]
/pj-rehearse pull-ci-openshift-operator-framework-catalogd-test-unit

**Comment by ncdc** [comment]
/pj-rehearse pull-ci-openshift-operator-framework-catalogd-test-verify

**Comment by ncdc** [comment]
/pj-rehearse pull-ci-openshift-operator-framework-catalogd-test-unit

**Comment by ncdc** [comment]
/pj-rehearse pull-ci-openshift-operator-framework-catalogd-test-unit

**Comment by ncdc** [comment]
/pj-rehearse pull-ci-openshift-operator-framework-catalogd-test-unit

**Comment by ncdc** [comment]
/pj-rehearse pull-ci-openshift-operator-framework-catalogd-main-verify-deps

**Comment by ncdc** [comment]
/pj-rehearse pull-ci-openshift-operator-framework-catalogd-main-verify-deps

**Comment by ncdc** [comment]
/pj-rehearse pull-ci-openshift-operator-framework-catalogd-main-verify-deps

File: ci-operator/config/openshift/operator-framework-catalogd/openshift-operator-framework-catalogd-main.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+build_root:
+  from_repository: true
+images:
+- dockerfile_path: openshift/Dockerfile
+  from: base
+  inputs:
+    ocp_builder_rhel-8-golang-1.19-openshift-4.14:
+      as:
+      - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.19-openshift-4.14
+  to: catalogd-controller
```
Comment (ncdc) [inline ci-operator/config/openshift/operator-framework-catalogd/openshift-operator-framework-catalogd-main.yaml:19]: Is this an ok image name?

Comment (joelanford) [reply ci-operator/config/openshift/operator-framework-catalogd/openshift-operator-framework-catalogd-main.yaml:19]: In one of those recent upstream commits, I renamed to just "catalogd" for now. Shall we do that here as well? We can always expand back out if upstream adds more images.

Comment (ncdc) [reply ci-operator/config/openshift/operator-framework-catalogd/openshift-operator-framework-catalogd-main.yaml:19]: Sure, I'll make that change

**Review by joelanford — APPROVED** [review]
Looks reasonable to me!

/approve

**Comment by ncdc** [comment]
/hold

Going to make some changes

**Review by grokspawn — APPROVED** [review]

**Comment by dhellmann** [comment]
The new image name looks OK, it's consistent with the `olm-rukpak` image.

/approve

**Pull request merged**